module3/ex7_apply_to_file/ex7_apply_to_file.py

```diff
@@ -5,8 +5,32 @@ def file_length(filename):
     return os.stat(filename).st_size
 
 
-def filefunc():
-    pass
+def filefunc(dir_path, os_command):
+    success = {}
+    failure = {}
+    obj = os.scandir(dir_path)
+    for entry in obj:
+        if entry.is_file():
+            try:
+                success[entry.name] = os_command(entry)
+            except Exception as e:
+                failure[entry.name] = e
+
+    return success, failure
+
+
+def filefunc_recursively(dir_path, os_command):
+    success_rec = {}
+    failure_rec = {}
+    obj = os.walk(dir_path, topdown=True)
+    for root, dirs, files in obj:
+        for entry in files:
```

Comment:
It's good to apply general best practices for variables naming. Use the same noun in singular for the iterator/loop variable and in plural for the sequence variable. `for file in files:` in this case.